rust/src/tx_builder.rs

```diff
@@ -455,12 +462,17 @@ impl TransactionBuilder {
             certs: self.certs.clone(),
             withdrawals: self.withdrawals.clone(),
             update: None,
-            metadata_hash: match &self.metadata {
+            auxiliary_data_hash: match &self.auxiliary_data {
                 None => None,
-                Some(x) => Some(utils::hash_metadata(x)),
+                Some(x) => Some(utils::hash_auxiliary_data(x)),
             },
             validity_start_interval: self.validity_start_interval,
             mint: self.mint.clone(),
+            // TODO: update for use with Alonzo
+            script_data_hash: None,
+            collateral: None,
+            required_signers: None,
+            network_id: None,
```

Comment:
@rooooooooob , there must be setters added for these new fields, right?

Reply:
Maybe, I just updated the bare minimum in the tx builder as those are all optional fields. You can still manually set them by manual building tx. I'm not sure about the tx creation process for all alonzo-era txs so I left it like this.